main.py

```diff
@@ -0,0 +1,58 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from database import SessionLocal
+import crud
+import schemas
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/author/", response_model=list[schemas.Author])
+def get_author(db: Session = Depends(get_db)):
+    return crud.get_all_authors(db)
+
+
+@app.post("/author/", response_model=schemas.Author)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Author already exists",
+        )
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/book/", response_model=list[schemas.Book])
+def get_book(db: Session = Depends(get_db)):
+    return crud.get_all_books(db)
+
+
+@app.get("/book/{book_id}", response_model=schemas.Book)
+def read_single_book(book_id: int, db: Session = Depends(get_db)):
+    db_book = crud.get_book(db=db, book_id=book_id)
```

Comment:
The `crud.get_book` function is called here, but it is not defined in `crud.py`. Ensure that this function is implemented in `crud.py` to avoid runtime errors.